in_toto/runlib.py

```diff
@@ -315,20 +318,26 @@ def create_link_metadata(link_name, materials_dict=None, products_dict=None,
   if not byproducts:
     byproducts = {}
 
+  if record_working_directory:
+    working_directory = os.getcwd()
+  else:
+    working_directory = ""
+
   link_dict = {
     "name" : link_name,
     "materials" : materials_dict,
     "products" : products_dict,
     "command" : link_cmd_args,
     "byproducts" : byproducts,
-    "return_value" : return_value
+    "return_value" : return_value,
+    "working_directory" : working_directory
```

Comment:
I can make it conditional in link.py. Only add working_directory if its not empty.